File: src/main/java/net/skhu/dto/OdsayRouteResponse.java
package net.skhu.dto;

import lombok.Data;

@Data
public class OdsayRouteResponse {
    private int routeNumber;             // 경로 번호
    private int totalTime;               // 총 소요 시간 (분)
    private int payment;                 // 총 비용 (원)
    private int busTransitCount;         // 버스 환승 횟수
    private int subwayTransitCount;      // 지하철 환승 횟수
    private int totalBusTime;            // 총 버스 이동 시간
    private int totalSubwayTime;         // 총 지하철 이동 시간
    private int totalWalkTime;           // 총 도보 이동 시간
}

File: frontend/lib/models/odsay_route_response.dart
class OdsayRouteResponse {
  final int routeNumber;
  final int totalTime;
  final int payment;
  final int busTransitCount;
  final int subwayTransitCount;
  final int totalBusTime;
  final int totalSubwayTime;
  final int totalWalkTime;

  OdsayRouteResponse({
    required this.routeNumber,
    required this.totalTime,
    required this.payment,
    required this.busTransitCount,
    required this.subwayTransitCount,
    required this.totalBusTime,
    required this.totalSubwayTime,
    required this.totalWalkTime,
  });

  /// 버스 환승과 지하철 환승을 합산한 총 환승 횟수
  int get totalTransitCount => busTransitCount + subwayTransitCount;

  factory OdsayRouteResponse.fromJson(Map<String, dynamic> json) {
    return OdsayRouteResponse(
      routeNumber: json['routeNumber'],
      totalTime: json['totalTime'],
      payment: json['payment'],
      busTransitCount: json['busTransitCount'],
      subwayTransitCount: json['subwayTransitCount'],
      totalBusTime: json['totalBusTime'],
      totalSubwayTime: json['totalSubwayTime'],
      totalWalkTime: json['totalWalkTime'],
    );
  }
}